ffi/examples/read-table/read_table.c

```diff
@@ -167,6 +168,199 @@ void free_partition_list(PartitionList* list) {
   free(list);
 }
 
+
+// Our schema code might add (is nullable: x) to names, which we need to strip off.
+// Returns the location where the edit was made if one was made, or NULL if no edit
+char* fixup_name(char* name) {
+  char *s = name;
+  char found_paren = 0;
+  while (*s) {
+    if (*s == '(') {
+      *(s-1) = '\0';
+      found_paren = 1;
+      break;
+    }
+    s++;
+  }
+  if (found_paren) {
+    return s-1;
+  } else {
+    return NULL;
+  }
+}
+
+// This function looks at the type field in the schema to figure out which visitor to call. It's a
+// bit gross as the schema code is string based, a real implementation would have a more robust way
+// to represent a schema.
+uintptr_t visit_schema_item(SchemaItem* item, KernelSchemaVisitorState *state, CSchema *cschema) {
+  print_diag("Visiting schema item %s (%s)\n", item->name, item->type);
+  KernelStringSlice name = { item->name, strlen(item->name) };
+  ExternResultusize visit_res;
+  if (strcmp(item->type, "string") == 0) {
+    visit_res = visit_field_string(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "integer") == 0) {
+    visit_res = visit_field_integer(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "short") == 0) {
+    visit_res = visit_field_short(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "byte") == 0) {
+    visit_res = visit_field_byte(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "long") == 0) {
+    visit_res = visit_field_long(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "float") == 0) {
+    visit_res = visit_field_float(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "double") == 0) {
+    visit_res = visit_field_double(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "boolean") == 0) {
+    visit_res = visit_field_boolean(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "binary") == 0) {
+    visit_res = visit_field_binary(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "date") == 0) {
+    visit_res = visit_field_date(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "timestamp") == 0) {
+    visit_res = visit_field_timestamp(state, name, false, allocate_error);
+  } else if (strcmp(item->type, "timestamp_ntz") == 0) {
+    visit_res = visit_field_double(state, name, false, allocate_error);
```

Comment:
should this be a timestampntz visitor?
```suggestion
    visit_res = visit_field_timestamp_ntz(state, name, false, allocate_error);
```